tasks/main.yml

```diff
@@ -14,74 +14,31 @@
 #     See the License for the specific language governing permissions and
 #     limitations under the License.
 #
-- name: Check for 'insights-client' RPM availability
-  yum:
-    list: insights-client
-  register: yum_list_task
-  become: true
-
-- name: Install 'insights-client' if it is available
+- name: Install 'insights-client'
   yum:
     name: insights-client
-  when: yum_list_task.results|length > 0
   become: true
 
 - name: Set Insights Config Name to 'insights-client'
   set_fact:
     insights_name: 'insights-client'
-  when: yum_list_task.results|length > 0
-  become: true
-
-- name: Install 'redhat-access-insights' if 'insights-client' is not available
-  yum:
-    name: redhat-access-insights
-  when: yum_list_task.results|length == 0
-  become: true
-
-- name: Set Insights Config Name to 'redhat-access-insights'
-  set_fact:
-    insights_name: 'redhat-access-insights'
-  when: yum_list_task.results|length == 0
   become: true
 
-- name: Configure username in Insights' Config file
-  ini_file:
-    path: /etc/{{ insights_name }}/{{ insights_name }}.conf
-    section: "{{ insights_name }}"
-    option: username
-    value: "{{ redhat_portal_username }}"
-    no_extra_spaces: true
-    state: "{{ 'present' if redhat_portal_username else 'absent' }}"
-  when: redhat_portal_username is defined
-  become: true
-
-- name: Configure password in Insights Config file
-  ini_file:
-    path: /etc/{{ insights_name }}/{{ insights_name }}.conf
-    section: "{{ insights_name }}"
-    option: password
-    value: "{{ redhat_portal_password }}"
-    no_extra_spaces: true
-    state: "{{ 'present' if redhat_portal_username else 'absent' }}"
-  when: redhat_portal_username is defined
-  become: true
-
-- name: Check status of Insights .register file
-  stat: path=/etc/{{ insights_name }}/.registered
-  become: true
-  register: reg_file_task
-
-- name: Unregister if we are setting the display_name, and we have already registered
-  command: "{{ insights_name }} --unregister"
-  when:
-    - insights_display_name is defined
-    - reg_file_task.stat.exists == true
+- name: Set Insights Configuration Values
+  insights_config:
+    insights_name: 'insights-client'
+    username: "{{ redhat_portal_username }}"
+    password: "{{ redhat_portal_password }}"
+    auto_config: "{{ auto_config }}"
+    authmethod: "{{ authmethod }}"
   become: true
 
-- name: Register to the Red Hat Access Insights Service if necessary
-  command: "{{ insights_name }} --register {{ '--display-name='+insights_display_name if insights_display_name is defined else '' }} creates=/etc/{{ insights_name }}/.registered"
+- name: Register Insights Client
+  insights_registration:
+    insights_name: "{{ insights_name }}"
+    display_name: "{{ insights_display_name if insights_display_name is defined else '' }}"
```

Comment:
Might be appropriate to use the `|omit()` filter here?  Take a look at https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#omitting-parameters

Reply:
:+1: